torchchat/generate.py

```diff
@@ -3,218 +3,35 @@
-    top_k: int = 200
-    temperature: float = 0.0  # deterministic argmax if 0.0
-    compile: bool = False
-    compile_prefill: bool = False
-    speculate_k: int = 5
-    sequential_prefill: bool = False
-    max_autotune: bool = False
-    # (Misnomer) See Issue: https://github.com/pytorch/torchchat/issues/1273
-    is_torchtune_model: bool = False
-
-    def __post_init__(self):
-        if self.compile_prefill and self.sequential_prefill:
-            raise RuntimeError("prefill compilation requires parallel prefill")
-
-    def validate_build(
-        self, builder_args: BuilderArgs, model_description: str = "model"
-    ):
-        reason = ""
-        model_type = ""
-        if not self.sequential_prefill:
-            reason = "parallel prefill"
-        if self.compile_prefill:
-            reason = "model compilation for prefill"
-        if self.compile:
-            reason = "model compilation"
-        if builder_args.aoti_package_path:
-            model_type = "PT2"
-        if builder_args.dso_path:
-            model_type = "DSO"
-        if builder_args.pte_path:
-            model_type = "PTE"
-        if model_type and reason:
-            raise RuntimeError(
-                f"cannot perform {reason} because a {model_type} {model_description} is used"
-            )
-
-    @classmethod
-    def from_args(cls, args):
-        dso_path = getattr(args, "dso_path", None)
-        pte_path = getattr(args, "pte_path", None)
-        aoti_package_path = getattr(args, "aoti_package_path", None)
-        sequential_prefill = (
-            args.sequential_prefill or bool(aoti_package_path) or bool(pte_path) or bool(dso_path)
-        )
-
-        # Validate that all image prompts exist before expensive model load
-        if image_prompts := getattr(args, "image_prompts", None):
-            non_existent_image_prompts = [
-                image_prompt
-                for image_prompt in image_prompts
-                if (not os.path.exists(image_prompt))
-            ]
-            if non_existent_image_prompts:
-                raise RuntimeError(
-                    f"Image prompt {non_existent_image_prompts} does not exist"
-                )
-
-        return cls(
-            prompt=getattr(args, "prompt", ""),
-            encoded_prompt=None,
-            image_prompts=image_prompts,
-            chat_mode=args.chat,
-            gui_mode=args.gui,
-            num_samples=getattr(args, "num_samples", 1),
-            max_new_tokens=args.max_new_tokens,
-            top_k=args.top_k,
-            temperature=args.temperature,
-            compile=args.compile,
-            compile_prefill=args.compile_prefill,
-            speculate_k=args.speculate_k,
-            sequential_prefill=sequential_prefill,
-            max_autotune=args.max_autotune,
-            is_torchtune_model=args.model and args.model.endswith("tune"),
-        )
-
-
-class Generator:
+class SingleGPUGenerator(Generator):
```

Comment:
I assume this name was chosen as contrast to distributed multi-GPU? There's nothing specific to GPU here, and the same generator is also used for ET models, CPU and accelerators.  How about `LocalGenerator` for the base case that's just running on the present node's resources, whatever they may be?

Reply:
Good point! Thanks @mikekgfb

Reply:
Fixed